### PR TITLE
Feature/simulator

### PR DESCRIPTION
**Pull request by dehli**
# PR for Issue #183.

## PR Status

**READY**

## Changes proposed in the pull request:
- Adds tabbed-container
- Adds active line class
- Adds state and command to simulator
- **Note** to prevent this PR from getting to big these will addressed in a future PR:
    - There are obviously lots of styling improvements that need to be done. I'll have another PR where I improve the way it looks and also componentize some more elements (similar to what I did with `active-line`)
    - I'm also going to change tabbed-container so it's just the tab bar, and then use that component on the game page.

## Breaking changes?
- None

## Screenshot of feature:
image

## Demo
http://deploy-preview-187--dev-wombats.netlify.com/

@emilyseibert @oconn @dehli

### CODE REVIEW

File: less/components/tabbed-container.less [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    .content {
+        position: relative;
+        flex-grow: 1;
+    }
+
+    .tabs {
+        display: flex;
+        
+        button {
```
Comment (emilyseibert) [inline less/components/tabbed-container.less]: classname

File: less/panels/simulator.less [hunk 2]
```diff
@@ -8,13 +8,31 @@
...
-            // TODO: Pull the editor into its own component (probably will switch to Ace)
-            font-family: monospace;
-            border: none;
-            width: calc(~"100% - " 2 * @editor-padding);
+        .tabbed-container {
             height: @canvas-dimensions;
+
+            #editor {
+                @editor-padding: 2px;
+                // TODO: Pull the editor into its own component (probably will switch to Ace)
```
Comment (emilyseibert) [inline less/panels/simulator.less]: is there an issue for that?

File: less/panels/simulator.less [hunk 3]
```diff
@@ -8,13 +8,31 @@
...
+                border: none;
+                width: calc(~"100% - " 2 * @editor-padding);
+                height: calc(~"100% - " 2 * @editor-padding);
+            }
+
+            .output {
+                color: @white;
+                padding: 10px;
+
+                div {
```
Comment (emilyseibert) [inline less/panels/simulator.less]: classname

File: less/panels/simulator.less [hunk 4]
```diff
@@ -8,13 +8,31 @@
...
+                    margin-bottom: 10px;
+                    
+                    h3 {
```
Comment (emilyseibert) [inline less/panels/simulator.less]: classname

File: src/cljs/wombats_web_client/components/tabbed_container.cljs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                                         "active-line-top")
+                           :onClick #(on-index-change tab-index)} 
+                  label])
+               tabs))
+
+(defn- render [{:keys [tabs index on-index-change] :as props}]
+  (let [render (:render (tabs index))]
+    [:div {:class-name "tabbed-container"}
+     
+     [:div {:class-name "content"}
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/tabbed_container.cljs]: `:div.content`

File: src/cljs/wombats_web_client/components/tabbed_container.cljs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+               tabs))
+
+(defn- render [{:keys [tabs index on-index-change] :as props}]
+  (let [render (:render (tabs index))]
+    [:div {:class-name "tabbed-container"}
+     
+     [:div {:class-name "content"} 
+      [render]]
+     
+     [:div {:class-name "tabs"}
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/tabbed_container.cljs]: `div.tabs`

File: src/cljs/wombats_web_client/panels/simulator.cljs [hunk 7]
```diff
@@ -37,58 +47,83 @@
...
     (arena/arena arena-frame canvas-id)
     [:div {:class-name "left-pane"}
      [:canvas {:id canvas-id
                :width dimensions
                :height dimensions}]]))
 
-(defn- render-right-pane [state wombats]
+(defn- render-code-tab [state]
+  [:textarea {:id "editor"
+              :onChange #(on-code-change! %)
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/panels/simulator.cljs]: `:on-change`

File: src/cljs/wombats_web_client/panels/simulator.cljs [hunk 8]
```diff
@@ -37,58 +47,83 @@
...
+              :value (or (get-player-code state) "")}])
+
+(defn- render-output-tab [sim-state]
+  [:div {:class-name "output"}
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/panels/simulator.cljs]: `:div.output`

File: src/cljs/wombats_web_client/panels/simulator.cljs [hunk 9]
```diff
@@ -37,58 +47,83 @@
...
+
+(defn- render-right-pane [cmpnt-state sim-state]
   [:div {:class-name "right-pane"}
-   [:textarea 
-    {:id "editor"
-     :onChange #(on-code-change! %)
-     :value (or (get-player-code state) "")}]
-   [:button {:onClick #(on-step-click! % state)}
+   (render-tabbed-container cmpnt-state sim-state)
+   [:button {:onClick #(on-step-click! % sim-state)}
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/panels/simulator.cljs]: `:on-click`

File: less/panels/simulator.less (hunk 2 above)
Comment (dehli) [inline less/panels/simulator.less]: Yes #188. I'll add it in the comment

**Pull request merged**